### DISCUSSION TITLE
Multiple so-sensor-clean instances running

### DISCUSSION THREAD

**Discussion by 0p71k** [Security-Onion-Solutions/securityonion#2727] [Unsupported Versions] [answered]
Hi everyone! For about 2 weeks, I'm having issue with my stand-alone SO instance. Yesterday, I found the issue: I have a lot of "/bin/bash /usr/sbin/so-sensor-clean" running. Right now, it's 89 different processes. As you can imagine, it's really killing my machine. The load average is 150 right now, but it was over 750 yesterday. If I "sudo pkill sensor-clean", it solves the issue for a few minutes. 

So, I took a look the that Bash script and found at the end that it's specifically checking the number of running processes to prevent that behaviour. I added a line to the script to understand the issue: "echo $IS_RUNNING >>$LOG" and I can see in the log file that the variable is always 0 explaining why it keeps starting a new instance. When I run the "ps aux | grep sensor-clean | ..." command manually, I get the real number of processes but when I run it inside a Bash script, it doesn't work correctly.

I was running 2.3.10 but I upgraded to 2.3.21 yesterday but the issue persists.

Do you guys have any clue to help me resolve that issue?

Thanks a lot!

**Comment by trb143** [comment, marked as answer]
This has been fixed in merge https://github.com/Security-Onion-Solutions/securityonion/commit/944817732b54eb640bab941d1f87da066de1178d

**Reply by m0duspwnens** [reply]
@0p71k, this is the change needed to be made to resolve the issue. The file can be found on the manager node at `/opt/so/saltstack/default/salt/common/tools/sbin/so-sensor-clean`

After this change is made, you can reboot to clean up all the processes or pkill them then run a `sudo salt-call state.highstate` so that the script on the node can update.

**Reply by 0p71k** [reply]
Thanks @trb143 and @m0duspwnens! I can confirm that the issue is resolved after manually editing the file as per the commit. Thank you guys for the quick reply and keep up the good work!
